BALSAMIC/assets/scripts/sex_prediction_tga.py

```diff
@@ -0,0 +1,487 @@
+
+    Args:
+        df (pd.DataFrame): DataFrame containing coverage data.
+
+    Returns:
+        pd.DataFrame: DataFrame with statistics such as mean, median, min, max, etc.
+    """
+    return (
+        df.groupby("chrom")["depth"]
+        .agg(
+            mean="mean", median="median", min="min", max="max", std="std", count="count"
+        )
+        .reset_index()
+    )
+
+
+def extract_stats(stats: pd.DataFrame) -> Dict[str, float]:
+    """Extract statistics from a DataFrame into a dictionary.
+
+    Args:
+        stats (pd.DataFrame): DataFrame with statistical summaries.
+
+    Returns:
+        Dict[str, float]: Dictionary containing extracted statistics.
+    """
+
+    stats: dict = stats.T.to_dict()
+    sample_dict = {}
+    sample_dict["X_mean"] = stats[0]["mean"]
+    sample_dict["X_median"] = stats[0]["median"]
+    sample_dict["X_min"] = stats[0]["min"]
+    sample_dict["X_max"] = stats[0]["min"]
+    sample_dict["X_std"] = stats[0]["std"]
+    sample_dict["X_count"] = stats[0]["count"]
+    sample_dict["Y_mean"] = stats[1]["mean"]
+    sample_dict["Y_median"] = stats[1]["median"]
+    sample_dict["Y_min"] = stats[1]["min"]
+    sample_dict["Y_max"] = stats[1]["max"]
+    sample_dict["Y_std"] = stats[1]["std"]
+    sample_dict["Y_count"] = stats[1]["count"]
+    return sample_dict
+
+
+def retrieve_file_info(filepath: str) -> (str, str):
+    """Extract sample name and CNN type from file name."""
+    filename: str = os.path.basename(filepath)
+    sample_name: str = filename.split(".")[0]
+    cnn_type: str = "antitarget" if "antitarget" in filename else "target"
+    return sample_name, cnn_type
+
+
+def get_predicted_sex(y_x_frac: float) -> Dict[str, str]:
+    """Predict sex and confidence based on the Y/X ratio.
+
+    Args:
+        y_x_frac (float): Ratio of Y to X.
+
+    Returns:
+        Dict[str, str]: Predicted sex and confidence level.
+    """
+    thresholds = {
+        "male_high": 0.9,
+        "male_medium": 0.7,
+        "male_low": 0.2,
+        "female_high": 0.00001,
+        "female_medium": 0.05,
+        "female_low": 0.15,
+    }
+
+    if y_x_frac > thresholds["male_low"]:
+        if y_x_frac >= thresholds["male_high"]:
+            return {"predicted_sex": "male", "confidence": "high"}
+        elif y_x_frac >= thresholds["male_medium"]:
+            return {"predicted_sex": "male", "confidence": "medium"}
+        else:
+            return {"predicted_sex": "male", "confidence": "low"}
+    elif y_x_frac <= thresholds["female_low"]:
+        if y_x_frac >= thresholds["female_medium"]:
+            return {"predicted_sex": "female", "confidence": "medium"}
+        elif y_x_frac >= thresholds["female_high"]:
+            return {"predicted_sex": "female", "confidence": "high"}
+        else:
+            return {"predicted_sex": "female", "confidence": "low"}
+    else:
+        return {"predicted_sex": "unknown", "confidence": "low"}
```

Comment:
do we really need nested conditionals or can we simplify it? Also, can we remove the thresholds dict since it's only used one and makes it harder to read? (the information about the confidence and the sex is specified in the return anyways)

```suggestion
    if y_x_frac >= 0.9:
        return {"predicted_sex": "male", "confidence": "high"}
    elif y_x_frac <= 0.00001:
        return {"predicted_sex": "female", "confidence": "high"}
    elif y_x_frac >= 0.7:
        return {"predicted_sex": "male", "confidence": "medium"}
    elif y_x_frac <= 0.05:
        return {"predicted_sex": "female", "confidence": "medium"}
    elif y_x_frac > 0.2:
        return {"predicted_sex": "male", "confidence": "low"}
    elif y_x_frac <= 0.15:
        return {"predicted_sex": "female", "confidence": "low"}
    else:
        return {"predicted_sex": "unknown", "confidence": "low"}
```

Reply:
Done!